presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp

```diff
@@ -69,47 +69,116 @@ class PartitionAndSerializeOperator : public Operator {
     return !input_;
   }
 
+  // When a new input is added to the operator, we need to reset a few
+  // counters/states.
   void addInput(RowVectorPtr input) override {
     input_ = std::move(input);
-  }
-
-  RowVectorPtr getOutput() override {
-    if (!input_) {
-      return nullptr;
+    numInput = input_->size();
+    if (replicateNullsAndAny_) {
+      collectNullRows();
     }
-    collectNullRows();
+    thisNullCount = nullRows_.countSelected();
+    totalNullCount += thisNullCount;
```

Comment:
Would it make sense to move this to collectNullRows?

Reply:
moved.